cmd/pipelines/logs.go

```diff
@@ -0,0 +1,142 @@
+package pipelines
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/databricks/cli/cmd/root"
+	"github.com/databricks/cli/libs/cmdctx"
+	"github.com/databricks/cli/libs/cmdgroup"
+	"github.com/databricks/cli/libs/cmdio"
+	"github.com/databricks/databricks-sdk-go/service/pipelines"
+	"github.com/spf13/cobra"
+)
+
+// Finds the update with the most recent CreationTime from a list of updates.
+func getMostRecentUpdateId(updates []pipelines.UpdateInfo) (string, error) {
+	if len(updates) == 0 {
+		return "", errors.New("no updates provided")
+	}
+
+	var mostRecentUpdate *pipelines.UpdateInfo
+	var mostRecentTime int64 = 0
+
+	for i := range updates {
+		update := &updates[i]
+		if update.CreationTime > mostRecentTime {
+			mostRecentTime = update.CreationTime
+			mostRecentUpdate = update
+		}
+	}
+
+	if mostRecentUpdate == nil {
+		return "", errors.New("no valid updates found")
+	}
+
+	return mostRecentUpdate.UpdateId, nil
+}
+
+// Creates a SQL filter condition for a field with multiple possible values,
+// generating "field in ('value1')" for a single value or "field in ('value1', 'value2')" for multiple values.
+func buildFieldFilter(field string, values []string) string {
+	if len(values) == 0 {
+		return ""
+	}
+
+	quotedValues := "'" + strings.Join(values, "', '") + "'"
+	return fmt.Sprintf("%s in (%s)", field, quotedValues)
+}
+
+// Cconstructs a SQL filter string for pipeline events based on the provided parameters.
+func buildPipelineEventFilter(updateId string, levels, eventTypes []string) string {
+	var filterParts []string
+
+	if updateId != "" {
+		filterParts = append(filterParts, fmt.Sprintf("update_id = '%s'", updateId))
+	}
+
+	if levelFilter := buildFieldFilter("level", levels); levelFilter != "" {
+		filterParts = append(filterParts, levelFilter)
+	}
+
+	if typeFilter := buildFieldFilter("event_type", eventTypes); typeFilter != "" {
+		filterParts = append(filterParts, typeFilter)
+	}
+
+	if len(filterParts) > 0 {
+		return strings.Join(filterParts, " AND ")
+	}
+
+	return ""
+}
+
+func logsCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "logs [flags] PIPELINE_ID",
+		Args:  root.ExactArgs(1),
+		Short: "Retrieve events for a pipeline",
+		Long: `Retrieve events for the pipeline identified by PIPELINE_ID.
+Show events for the pipeline's most recent update by default.
+
+Example usage:
+  1. pipelines logs my-pipeline --update-id update-1
```

Comment:
Documentation and code suggest that we accept pipeline_id, but code examples are using a resource name. It's troublesome to copy-paste pipeline_id. We should have a UX similar to other commands on pipelines.

We can make it work for an arbitrary pipeline_id, but I find it secondary